cmd/update-main.go

```diff
@@ -17,274 +17,478 @@
 package cmd
 
 import (
-	"bytes"
+	"crypto"
+	"encoding/hex"
 	"encoding/json"
 	"fmt"
 	"io/ioutil"
 	"net/http"
 	"os"
-	"os/exec"
+	"path/filepath"
 	"runtime"
 	"strings"
 	"time"
 
 	"github.com/fatih/color"
+	"github.com/inconshreveable/go-update"
+	isatty "github.com/mattn/go-isatty"
 	"github.com/minio/cli"
-	"github.com/minio/mc/pkg/console"
 	"github.com/minio/mc/pkg/probe"
+	_ "github.com/minio/sha256-simd" // Needed for sha256 hash verifier.
+	"github.com/segmentio/go-prompt"
 )
 
 // Check for new software updates.
 var updateCmd = cli.Command{
 	Name:   "update",
-	Usage:  "check for new software updates",
+	Usage:  "update mc to latest release",
 	Action: mainUpdate,
-	Before: setGlobalsFromContext,
 	Flags: []cli.Flag{
 		cli.BoolFlag{
 			Name:  "quiet, q",
-			Usage: "suppress chatty console output",
+			Usage: "disable any update prompt message",
 		},
 		cli.BoolFlag{
 			Name:  "json",
 			Usage: "enable JSON formatted output",
 		},
 	},
 	CustomHelpTemplate: `Name:
-  {{.HelpName}} - {{.Usage}}
+   {{.HelpName}} - {{.Usage}}
 
 USAGE:
-  {{.HelpName}}{{if .VisibleFlags}} [FLAGS]{{end}}
+   {{.HelpName}}{{if .VisibleFlags}} [FLAGS]{{end}}
 {{if .VisibleFlags}}
 FLAGS:
   {{range .VisibleFlags}}{{.}}
   {{end}}{{end}}
+EXIT STATUS:
+   0 - You are already running the most recent version.
```

Comment:
It is probably easier to merge exit status 0 and 1 to 0, if the user wants more information, he can pass --json and parse output.

Reply:
No it is not both needs to be differentiated - see the server implementation.